File: Leetcode/leetcode-0083-RemoveDuplicatesfromSortedList-[linkedlist]-[easy]/Solution.simpler.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        visited = -10000;
        cur = head
        prev = None
        while (cur != None):
            if (visited == -10000 or visited != cur.val):
                visited = cur.val
                prev = cur
                cur = cur.next
            else:
                cur = cur.next
                prev.next = cur
        
        return head

File: Leetcode/leetcode-0083-RemoveDuplicatesfromSortedList-[linkedlist]-[easy]/Solution.simpler.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        int visited = -1000;
        ListNode prev = null;
        ListNode cur = head;
        while (cur != null) {
            if (visited == -1000 || cur.val != visited) {
                visited = cur.val;
                prev = cur;
                cur = cur.next;
            } else {
                cur = cur.next;
                prev.next = cur;
            }
        }

        return head;
    }
}